### PR TITLE
[Ac 1563] Unable to load billing and subscription related pages for non-enterprise organizations

### PR DESCRIPTION
**Pull request by cyprain-okeke**
## Type of change

```
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

In CloudQA-US and CloudQA-EU, the following pages will not load when the subscription is not enterprise:

Organization Info

Subscription

Payment Method

## Code changes

* **file.ext:** Description of what was changed and why

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/Services/Implementations/OrganizationService.cs [hunk 1]
```diff
@@ -446,8 +446,8 @@ public async Task VerifyBankAsync(Guid organizationId, int amount1, int amount2)
             RevisionDate = DateTime.UtcNow,
             Status = OrganizationStatusType.Created,
             UsePasswordManager = true,
-            SmSeats = (short)(secretsManagerPlan.BaseSeats + signup.AdditionalSmSeats.GetValueOrDefault()),
-            SmServiceAccounts = secretsManagerPlan.BaseServiceAccount + signup.AdditionalServiceAccounts.GetValueOrDefault(),
+            SmSeats = secretsManagerPlan?.BaseSeats + signup.AdditionalSmSeats.GetValueOrDefault(),
+            SmServiceAccounts = secretsManagerPlan?.BaseServiceAccount + signup.AdditionalServiceAccounts.GetValueOrDefault(),
```
Comment (eliykat) [inline src/Core/Services/Implementations/OrganizationService.cs:450]: Same here, move to its own `if` block.

File: test/Core.Test/Services/OrganizationServiceTests.cs [hunk 2]
```diff
@@ -145,6 +145,58 @@ await sutProvider.GetDependency<IReferenceEventService>().Received(1)
...
+
+        var result = await sutProvider.Sut.SignUpAsync(signup);
+
+        await sutProvider.GetDependency<IOrganizationRepository>().Received(1).CreateAsync(
+            Arg.Is<Organization>(o =>
+                o.Seats == passwordManagerPlan.BaseSeats + signup.AdditionalSeats
+                && o.SmSeats == null
+                && o.SmServiceAccounts == null));
+        await sutProvider.GetDependency<IOrganizationUserRepository>().Received(1).CreateAsync(
+            Arg.Is<OrganizationUser>(o => o.AccessSecretsManager == signup.UseSecretsManager));
```
Comment (eliykat) [inline test/Core.Test/Services/OrganizationServiceTests.cs:170]: You should set `signup.UseSecretsManager` explicitly. It should always be `false` when signing up to families, but the autodata might set it to `true` here which would be an invalid state.

File: test/Core.Test/Services/OrganizationServiceTests.cs [hunk 3]
```diff
@@ -145,6 +145,58 @@ await sutProvider.GetDependency<IReferenceEventService>().Received(1)
...
+
+        await sutProvider.GetDependency<IReferenceEventService>().Received(1)
+            .RaiseEventAsync(Arg.Is<ReferenceEvent>(referenceEvent =>
+                referenceEvent.Type == ReferenceEventType.Signup &&
+                referenceEvent.PlanName == purchaseOrganizationPlan[0].Name &&
+                referenceEvent.PlanType == purchaseOrganizationPlan[0].Type &&
```
Comment (eliykat) [inline test/Core.Test/Services/OrganizationServiceTests.cs:176]: Isn't this the same as the `passwordManagerPlan` variable? Why do we need `purchaseOrganizationPlan`?

File: test/Core.Test/Services/OrganizationServiceTests.cs [hunk 4]
```diff
@@ -145,6 +145,58 @@ await sutProvider.GetDependency<IReferenceEventService>().Received(1)
...
+                referenceEvent.Seats == result.Item1.Seats &&
+                referenceEvent.Storage == result.Item1.MaxStorageGb));
```
Comment (eliykat) [inline test/Core.Test/Services/OrganizationServiceTests.cs:178]: You shouldn't use the `result` object from the sut to base assertions from - it's part of what you're trying to test. Instead, get these values from the `SignUp` object or another input.

File: test/Core.Test/Services/OrganizationServiceTests.cs [hunk 5]
```diff
@@ -145,6 +145,58 @@ await sutProvider.GetDependency<IReferenceEventService>().Received(1)
...
+        // TODO: add reference events for SmSeats and Service Accounts - see AC-1481
+
+        Assert.NotNull(result);
+        Assert.NotNull(result.Item1);
+        Assert.NotNull(result.Item2);
+        Assert.IsType<Tuple<Organization, OrganizationUser>>(result);
```
Comment (eliykat) [inline test/Core.Test/Services/OrganizationServiceTests.cs:184]: Can we assert the actual values of the `result` object?

File: test/Core.Test/Services/OrganizationServiceTests.cs [hunk 6]
```diff
@@ -145,6 +145,58 @@ await sutProvider.GetDependency<IReferenceEventService>().Received(1)
...
+
+        await sutProvider.GetDependency<IPaymentService>().Received(1).PurchaseOrganizationAsync(
+            Arg.Any<Organization>(),
+            signup.PaymentMethodType.Value,
+            signup.PaymentToken,
+            Arg.Is<List<Plan>>(plan => plan.All(p => purchaseOrganizationPlan.Contains(p))),
```
Comment (eliykat) [inline test/Core.Test/Services/OrganizationServiceTests.cs:190]: It should only have received the PM plan, not the SM plan. This assertion is not strict enough.

I suggest `plan.Single() == passwordManagerPlan`.

**Review by eliykat — CHANGES_REQUESTED** [review]

File: src/Core/Services/Implementations/OrganizationService.cs (hunk 1 above)
Comment (cyprain-okeke) [reply src/Core/Services/Implementations/OrganizationService.cs:450]: resolved

[the remark this answers is not in the document]
Comment (cyprain-okeke) [reply src/Core/Services/Implementations/OrganizationService.cs:450]: resolved

[the remark this answers is not in the document]
Comment (cyprain-okeke) [reply src/Core/Services/Implementations/OrganizationService.cs:450]: resolved

File: src/Core/OrganizationFeatures/OrganizationSubscriptions/UpgradeOrganizationPlanCommand.cs [hunk 7]
```diff
@@ -267,8 +267,8 @@ public class UpgradeOrganizationPlanCommand : IUpgradeOrganizationPlanCommand
         organization.PublicKey = upgrade.PublicKey;
         organization.PrivateKey = upgrade.PrivateKey;
         organization.UsePasswordManager = true;
-        organization.SmSeats = (short)(newSecretsManagerPlan.BaseSeats + upgrade.AdditionalSmSeats.GetValueOrDefault());
-        organization.SmServiceAccounts = newSecretsManagerPlan.BaseServiceAccount + upgrade.AdditionalServiceAccounts.GetValueOrDefault();
+        organization.SmSeats = newSecretsManagerPlan?.BaseSeats + upgrade.AdditionalSmSeats.GetValueOrDefault();
+        organization.SmServiceAccounts = newSecretsManagerPlan?.BaseServiceAccount + upgrade.AdditionalServiceAccounts.GetValueOrDefault();
```
[the remark this answers is not in the document]
Comment (cyprain-okeke) [reply src/Core/OrganizationFeatures/OrganizationSubscriptions/UpgradeOrganizationPlanCommand.cs:271]: resolved

File: src/Api/Models/Response/Organizations/OrganizationResponseModel.cs [hunk 8]
```diff
@@ -27,7 +27,11 @@ public OrganizationResponseModel(Organization organization, string obj = "organi
         BusinessTaxNumber = organization.BusinessTaxNumber;
         BillingEmail = organization.BillingEmail;
         Plan = new PlanResponseModel(StaticStore.PasswordManagerPlans.FirstOrDefault(plan => plan.Type == organization.PlanType));
-        SecretsManagerPlan = new PlanResponseModel(StaticStore.SecretManagerPlans.FirstOrDefault(plan => plan.Type == organization.PlanType));
+        var matchingPlan = StaticStore.SecretManagerPlans.FirstOrDefault(plan => plan.Type == organization.PlanType);
```
[the remark this answers is not in the document]
Comment (cyprain-okeke) [reply src/Api/Models/Response/Organizations/OrganizationResponseModel.cs:30]: resolved

File: test/Core.Test/Services/OrganizationServiceTests.cs (hunk 2 above)
Comment (cyprain-okeke) [reply test/Core.Test/Services/OrganizationServiceTests.cs:170]: yeah

File: test/Core.Test/Services/OrganizationServiceTests.cs (hunk 3 above)
Comment (cyprain-okeke) [reply test/Core.Test/Services/OrganizationServiceTests.cs:176]: you are right, is the same

File: test/Core.Test/Services/OrganizationServiceTests.cs (hunk 4 above)
Comment (cyprain-okeke) [reply test/Core.Test/Services/OrganizationServiceTests.cs:178]: We are testing that what we added to the organisation table is actually what we are Raising in the RaiseEventAsync.

File: test/Core.Test/Services/OrganizationServiceTests.cs (hunk 5 above)
Comment (cyprain-okeke) [reply test/Core.Test/Services/OrganizationServiceTests.cs:184]: This is a new signup, It means we have to assert all the properties of the organization object and the organizationuser

File: test/Core.Test/Services/OrganizationServiceTests.cs (hunk 6 above)
Comment (cyprain-okeke) [reply test/Core.Test/Services/OrganizationServiceTests.cs:190]: resolved

File: src/Core/OrganizationFeatures/OrganizationSubscriptions/UpgradeOrganizationPlanCommand.cs [hunk 9]
```diff
@@ -267,9 +267,14 @@ public class UpgradeOrganizationPlanCommand : IUpgradeOrganizationPlanCommand
...
-        organization.SmSeats = (short)(newSecretsManagerPlan.BaseSeats + upgrade.AdditionalSmSeats.GetValueOrDefault());
-        organization.SmServiceAccounts = newSecretsManagerPlan.BaseServiceAccount + upgrade.AdditionalServiceAccounts.GetValueOrDefault();
-        organization.UseSecretsManager = upgrade.UseSecretsManager;
+
+        if (upgrade.UseSecretsManager)
+        {
+            organization.SmSeats = newSecretsManagerPlan.BaseSeats + upgrade.AdditionalSmSeats.GetValueOrDefault();
+            organization.SmServiceAccounts = newSecretsManagerPlan.BaseServiceAccount.GetValueOrDefault() +
+                                             upgrade.AdditionalServiceAccounts.GetValueOrDefault();
+            organization.UseSecretsManager = upgrade.UseSecretsManager;
```
Comment (eliykat) [inline src/Core/OrganizationFeatures/OrganizationSubscriptions/UpgradeOrganizationPlanCommand.cs:276]: This should always be set (move to outside if the if statement)

File: src/Core/Services/Implementations/OrganizationService.cs [hunk 10]
```diff
@@ -446,11 +446,16 @@ public async Task VerifyBankAsync(Guid organizationId, int amount1, int amount2)
...
-            SmServiceAccounts = secretsManagerPlan.BaseServiceAccount + signup.AdditionalServiceAccounts.GetValueOrDefault(),
-            UseSecretsManager = signup.UseSecretsManager
         };
 
+        if (signup.UseSecretsManager)
+        {
+            organization.SmSeats = secretsManagerPlan.BaseSeats + signup.AdditionalSmSeats.GetValueOrDefault();
+            organization.SmServiceAccounts = secretsManagerPlan.BaseServiceAccount.GetValueOrDefault() +
+                                             signup.AdditionalServiceAccounts.GetValueOrDefault();
+            organization.UseSecretsManager = signup.UseSecretsManager;
```
Comment (eliykat) [inline src/Core/Services/Implementations/OrganizationService.cs:456]: This should always be set (move to outside if the if statement)

**Review by r-tome — APPROVED** [review]
Looks great!

**Pull request merged**